django_airavata/apps/api/views.py

```diff
@@ -303,13 +303,12 @@ def clone(self, request, experiment_id=None):
     @detail_route(methods=['post'])
     def cancel(self, request, experiment_id=None):
         try:
-            request.airavata_client.launchExperiment(
+            request.airavata_client.terminateExperiment(
                 request.authz_token, experiment_id, self.gateway_id)
             return Response({'success': True})
         except Exception as e:
             log.error("Cancel action has thrown the following error: ", e)
-            return Response({'success': False, 'errorMessage': e.message})
-
+            raise Http404(str(e)) from e
```

Comment:
Just do a `raise` here to re-raise the exception. exceptions.py will convert it into the write HTTP response.

Reply:
>  raise Http404(str(e)) from e

I have added the raise Http404 above. WIll this not work?

Reply:
Yeah, that works, but its not necessary since exceptions.py will convert it into an HTTP response. That way we consolidate the logic for turning exceptions into HTTP responses in one place.

Reply:
Okay. I have fixed it and removed the HTTP response.